openai/logger.py

```diff
@@ -0,0 +1,277 @@
+            fine_tunes = [fine_tune]
+
+        else:
+            # get list of fine_tune to log
+            fine_tunes = FineTune.list()
+            if not fine_tunes or fine_tunes.get("data") is None:
+                print("No fine-tune jobs have been retrieved")
+                return
+            fine_tunes = fine_tunes["data"][-n_jobs if n_jobs is not None else None :]
+
+        # log starting from oldest fine_tune
+        show_warnings = False if id is None and n_jobs is None else True
+        fine_tune_logged = [
+            cls._log_fine_tune(
+                fine_tune,
+                project,
+                entity,
+                force,
+                show_warnings,
+                **kwargs_wandb_init,
+            )
+            for fine_tune in fine_tunes
+        ]
+
+        if not show_warnings and not any(fine_tune_logged):
+            print("No new successful fine-tune were found")
+
+        return "🎉 wandb sync completed successfully"
+
+    @classmethod
+    def _log_fine_tune(
+        cls, fine_tune, project, entity, force, show_warnings, **kwargs_wandb_init
+    ):
+        fine_tune_id = fine_tune.get("id")
+        status = fine_tune.get("status")
+
+        # check run completed successfully
+        if show_warnings and status != "succeeded":
+            print(
+                f'Fine-tune job {fine_tune_id} has the status "{status}" and will not be logged'
+            )
+            return
+
+        # check run has not been logged already
+        run_path = f"{project}/{fine_tune_id}"
+        if entity is not None:
+            run_path = f"{entity}/{run_path}"
+        wandb_run = cls._get_wandb_run(run_path)
+        if wandb_run:
+            wandb_status = wandb_run.summary.get("status")
+            if show_warnings:
+                if wandb_status == "succeeded":
+                    print(
+                        f"Fine-tune job {fine_tune_id} has already been logged successfully at {wandb_run.url}"
+                    )
+                    if not force:
+                        print(
+                            'Use "--force" in the CLI or "force=True" in python if you want to overwrite previous run'
+                        )
+                else:
+                    print(
+                        f"A run for fine-tune job {fine_tune_id} was previously created but didn't end successfully"
+                    )
+                if wandb_status != "succeeded" or force:
+                    print(
+                        f"A new wandb run will be created for fine-tune job {fine_tune_id} and previous run will be overwritten"
+                    )
+            if wandb_status == "succeeded" and not force:
+                return
+
+        # retrieve results
+        results_id = fine_tune["result_files"][0]["id"]
+        results = File.download(id=results_id).decode("utf-8")
+
+        # start a wandb run
+        wandb.init(
+            job_type="finetune",
```

Comment:
Nit: Should this be `fine_tune` or `fine-tune` to be consistent with our API? (Does `finetune` mean something special within your system?)